tools/memap.py

```diff
@@ -330,12 +409,51 @@ def parse_map_file_armcc(self, file_desc):
             # Start decoding the map file
             for line in infile:
 
-                [name, size, section] = self.parse_section_armcc(line)
+                [object_name, object_size, section] = \
+                                self.parse_section_armcc(line)
 
-                if size == 0 or name == "" or section == "":
+                if object_size == 0 or object_name == "" or section == "":
                     pass
                 else:
-                    self.module_add(name, size, section)
+                    self.module_add(object_name, object_size, section)
+
+
+    def check_new_library_iar(self, line):
+        """
+        Searches for libraries and returns name. Example:
+        m7M_tls.a: [43]
+
+        """
+
+        rex_address_line = re.compile(r'^(.+\.a)\:.+$')
+
+        test_address_line = re.match(rex_address_line, line)
+
+        if test_address_line:
+            return test_address_line.group(1)
+        else:
+            return ""
+
+    def check_new_object_lib_iar(self, line):
+        """
+        Searches for objects within a library section and returns name. Example:
+        rt7M_tl.a: [44]
+            ABImemclr4.o                 6
+            ABImemcpy_unaligned.o      118
+            ABImemset48.o               50
+            I64DivMod.o                238
+            I64DivZer.o                  2
+
+        """
+
+        rex_address_line = re.compile(r'^\s+(.+\.o)\s.*')
```

Comment:
This comment is the same as the other Regular expression related ones:
1) `re` not `rex` is common in python
2) Please make this a Class or global constant